src/Common/src/Interop/Windows/kernel32/Interop.IsDebuggerPresent.cs

```diff
@@ -0,0 +1,14 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Runtime.InteropServices;
+
+internal partial class Interop
+{
+    internal partial class Kernel32
+    {
+        [DllImport(Libraries.Kernel32)]
+        internal static extern bool IsDebuggerPresent();
```

Comment:
We can leave these for now, as you've done a good job just translating what's there already, but I'm surprised these DebugBreak/IsDebuggerPresent P/Invokes are needed; I'd have expected the caller to just be able to use Debugger.Break and Debugger.IsAttached.

Reply:
Shall we remove the whole Dbg.cs? 

Its intention is to be used only in checked builds (are they even shipped?) and even then the primitives would only be supported on Windows (unless we change the code to use Debugger.Break). 

I would have voted for leaving it in place if it were designed to catch client misuse patterns. But this code was designed to catch internal cache implementation bugs during its development. Last time I remember using this ancient code was on the full .NET desktop framework 10+ years ago investigating cache corruptions due to ThreadAbortExceptions (which it is not designed to handle).

Reply:
> Shall we remove the whole Dbg.cs?

If you're ok with that (and it sounds like you are), then yes, I think we should :smile:

Reply:
What do you mean by removing the whole Dbg.cs? What about code that calls Dbg? There are about 170 references to the Dbg class.

Reply:
Yep, that would be the whole Dbg.cs. 
On the second thought, I would not get rid of all the asserts though - they might still be useful if there are ever bugs say on ARM64 or some other future platforms with weaker memory models. Just replace the Dbg.Assert with the System.Diagnostics.Debug.Assert.

Reply:
Ok, thanks. What about `Dbg.Trace`, `Dbg.Validate`, `Dbg.Dump`?

Reply:
Thanks for following up, @Marusyk! 
cc @StephenMolloy 

`Dbg.Trace`: out of all overrides I only see one being used: 
  &nbsp;&nbsp;&nbsp;&nbsp; `internal static void Trace(string tagName, string message)`
For now I think this can be replaced with `Debug.Write (string message, string category)` (use `category` for `tagName`). 

One problem is that some of the traces could be way too verbose (that's why there is a tag to filter them). Therefore, if we will ever need to trace extensively to investigate a timing-sensitive concurrency issue, we might need to reincarnate some DBG primitives (together with some lock free in-memory circular buffer, rather than using the debug output for investigating complex issues). However the amount of work to get to the bottom of those issues is likely going to be much more than just changing the tracing stuff.

`Dbg.Validate` and `Dbg.Dump`: I don't see them being used anywhere. The "DebugValidate" method `Validate` is going to try to call is gone from where it might have been used in the past and I am not sure what it should really be.

If you feel this needs to be split into a separate PR - feel free to do so.

Reply:
`Dbg.Validate` has 10 references (int [this](https://github.com/dotnet/corefx/blob/master/src/System.Runtime.Caching/src/System/Runtime/Caching/CacheExpires.cs#L153) file and [this one](https://github.com/dotnet/corefx/blob/master/src/System.Runtime.Caching/src/System/Runtime/Caching/CacheUsage.cs#L620)

Reply:
Ok, I see. Thank you for the explanation. So, what next? Should I return `Dbg.Assert` back or leave it as is?

Reply:
On one hand Dbg.Assert is pretty much a dead weight and can/should be replaced with Debug.Assert so this is a good cleanup. On the other hand I assume cleaning up all checked caching code was not the original purpose of this PR. @stephentoub what is the right thing to do?

Reply:
any news?

Reply:
Let me take a pass through it... looking now...

Reply:
I think we should remove the Validate stuff; it's only relevant in Debug builds, only applicable on Windows, and is currently a nop as pointed out anyway because of missing methods.  Since it's not currently adding any value, and even if it were addressed it would only add benefit (?) on one platform, and even then only in debug builds, I think it's more pain than it's worth.  I've pulled down the branch and can delete the relevant stuff; will push revisions up here for review and we can decide how to proceed.